core/src/test/java/org/apache/iceberg/catalog/CatalogTests.java

```diff
@@ -987,6 +987,13 @@ public void testListTables() {
     assertEmpty("Should not contain ns_2.table_1 after drop", catalog, ns2);
   }
 
+  @Test
+  public void listTablesForEmptyNamespace() {
+    Assertions.assertThatThrownBy(() -> catalog().listTables(Namespace.empty()))
+        .isInstanceOf(NoSuchNamespaceException.class)
+        .hasMessageMatching(".*Namespace does not exist.*|.*Invalid namespace.*");
```

Comment:
Empty namespaces are valid. We may need another test config like `requiresNamespaceCreate` to distinguish between catalogs that can store objects in the empty namespace and those that can't.

That would also solve the problem of different error messages.

Another option is to support the empty namespace in the REST catalog by always returning an empty set for `listTables` and `listNamespaces`.